doc/index.md

```diff
@@ -1061,25 +1058,24 @@ Create a new instance of an HTTP Server by passing in table `options`:
 
   - `.client_timeout`: Timeout (in seconds) to wait for client to send first bytes and/or complete TLS handshake (default: 10)
   - `.cq` (optional): A cqueues controller to use
-  - `.ctx`: an `openssl.ssl.context` object to use for tls connections. If `nil` is passed, a self-signed context will be generated
-  - `.family`: protocol family
-  - `.host`: address to bind to (required if not `.path`)
-  - `.port`: port to bind to (optional if tls isn't `nil`, in which case defaults to 80 for `.tls == false` or 443 if `.tls == true`)
-  - `.path`: path to UNIX socket (required if not `.host`)
-  - `.v6only`: allow ipv6 only (no ipv4-mapped-ipv6)
-  - `.mode`: fchmod or chmod socket after creating UNIX domain socket
-  - `.mask`: set and restore umask when binding UNIX domain socket
+  - `.ctx`: An `openssl.ssl.context` object to use for tls connections. If `nil` is passed, a self-signed context will be generated
+  - `.family`: (optional) Protocol family
+  - `.host`: Address to bind to (required if not `.path`)
+  - `.port`: Port to bind to (optional if tls isn't `nil`, in which case defaults to 80 for `.tls == false` or 443 if `.tls == true`)
+  - `.path`: Path to UNIX socket (required if not `.host`)
+  - `.v6only`: Allow ipv6 only (no ipv4-mapped-ipv6)
+  - `.mode`: (optional) fchmod or chmod socket after creating UNIX domain socket
+  - `.mask`: (optional) Set and restore umask when binding UNIX domain socket
+  - `.unlink`: (optional) Unlink socket path before binding
   - `.max_concurrent`: Maximum number of connections to allow live at a time (default: infinity)
-  - `.onerror`: function that will be called when an error occurs (default: do nothing)
-  - `.onstream`: callback function for handling a new client request
-  - `.unlink`: unlink socket path before binding?
-  - `.reuseaddr`: turn on SO_REUSEADDR flag?
-  - `.reuseport`: turn on SO_REUSEPORT flag?
-  - `.socket`: A cqueues socket object
-  - `.tls`: `nil`: allow both tls and non-tls connections
-  -         `true`: allows tls connections only
-  -         `false`: allows non-tls connections only
-  - `.version`: the http version to allow to connect (default: any)
+  - `.onerror`: Function that will be called when an error occurs (default: do nothing). See [server:onerror()](#http.server:onerror)
+  - `.onstream`: Callback function for handling a new client request. The function requires parameters for [*server*](#http.server) and [*stream*](#stream) and can throw an error without crashing the server instance. 
```

Comment:
> The function requires parameters for server and stream and can throw

Instead:

The function will receive the *server* object and a *stream* as parameters. If your function throws an error it will be reported from `:step()` (or `:loop()`).